docs/06-contributors/999-rfcs/2023-04-16-website-resource.md

```diff
@@ -0,0 +1,111 @@
+- **Stage Date**: 2023-04-16
+
+<!---
+Directions for above:
+
+Author(s): Git tag for PR authors and contributors
+Submission Date: Fill in with date of initial submission, YYYY-MM-DD
+Stage: Fill in with current stage in the RFC lifecycle
+Stage Date: Fill in with date of last stage change
+-->
+
+> Implementing the website resource following the [SDK spec](../../04-reference/wingsdk-spec.md#website).
+
+<!--
+This RFCs have 3 sections: Requirements -> Design -> Implementation. We intentionally start with *Design* since it
+is a great way to introduice the feature to readers.
+-->
+
+## Implementation
+
+### Implementation for aws
+
+Following a few AWS guides on the internet (brought in the end of the section), it seems that the common way to serve a static website is by using a bucket.
+
+After creating a public bucket, we'll upload the folder from the relative path.
+AWS requires a index document name (to serve on the "/" path).
+We'll start by hardcoding it to be "index.html", and can add it later to the website props upon request.
+
+The last thing will be a registration to the chosen domain (if exists).
+
+The website resource supports preflight code only.
+
+#### References:
+
+[Tutorial: Configuring a static website on Amazon S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html)
+[Use your domain for a static website in an Amazon S3 bucket](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/getting-started-s3.html)
+
+### Implementation for the simulator
+
+The website will be implemented by a static express server:
+
+```js
+const app = Express();
+
+app.use(express.static("<resolved path to the website files folder>"));
+
+app.listen(
+  "<port, ideally will be specified by the user, otherwise generated>"
+);
+```
+
+Since the website is served on the local machine, we cannot specify a domain, nor can we add a CDN while deploying to the simulator.
+
+### point for thought:
+
+I wonder if we should support .env files, in the future, (or any other kind of configuration-per-target system) for specifying different value for different targets:
+
+For example:
+
+In the wing file:
+
+```
+let main_website = new cloud.Website({path: "./website/main/dist", domain: MAIN_WEBSITE_DOMAIN})
+let back_office_website = new cloud.Website({path: "./website/back-office/dist", domain: BO_WEBSITE_DOMAIN})
+```
+
+In tf-aws.env/ prod.env:
+
+```
+MAIN_WEBSITE_DOMAIN= "my-company.com"
+BO_WEBSITE_DOMAIN= "bo.my-company.com"
+```
+
+In sim.env/ dev.env:
+
+```
+MAIN_WEBSITE_DOMAIN= "localhost:3000"
+BO_WEBSITE_DOMAIN= "localhost:4000"
+```
+
+This, of course, rises more questions then answers-
+how can we support different development environments? How do we tell wing which env file to use? Can we share those files between targets? (for example, running the same file for tf-gcp and tf-azure, and a different one for the sim).
+
+- Website props are brought here for reference:
+
+```ts
+struct WebsiteProps {
```

Comment:
We will likely also want to support 

- CORS
- deploying dynamic data into the website

See a real world example of how to setup a website using the aws cdk: https://github.com/cdklabs/construct-hub/blob/main/src/webapp/index.ts

A construct library: https://constructs.dev/packages/cdk-website/v/0.0.5?lang=typescript

Reply:
can you elaborate more on "deploying dynamic data into the website"? 
there are more configuration option that we may want to let the user decide on (regarding the cdn)
<img width="621" alt="image" src="https://user-images.githubusercontent.com/39455181/232441372-5711518f-9ecd-40d8-bf69-bc465ea48c44.png">
should I add some of them to the props too?